### PR TITLE
test: Assumeutxo: snapshots with less work should not be loaded

### PR DESCRIPTION
**Pull request by hernanmarino**
This PR adds a test which checks that snapshots with less accumulated work than the node's active chain, should not be loaded and return with an error. Although in a different context of discussion the missing test was detect in a thread in https://github.com/bitcoin/bitcoin/pull/29394 (see https://github.com/bitcoin/bitcoin/pull/29394#discussion_r1484122214)

### CODE REVIEW

**Comment by aureleoules** [comment]
> Looks like [corecheck.dev/bitcoin/bitcoin/pulls/29428](https://corecheck.dev/bitcoin/bitcoin/pulls/29428) is dead?

@maflcko thanks for the ping. The code I added to filter out flaky coverage was too slow and isn't very reliable so I removed it. Unfortunately, the UI will now display flaky coverage.

**Comment by hernanmarino** [comment]
> You'll have to remove the TODO on the top of the file as well, when the test was added?

Done. It'll be updated with my next push

**Review by BrandonOdiwuor — COMMENTED** [review]
Code Review ACK 86037415dd83da34d829f2f6bfb8dcbe08968180

Should remove the TODO fixed by this commit

**Comment by hernanmarino** [comment]
> Are you still working on this?

I haven´t been able to think of / work on other tests to add, but the current commit test is ready for review, so i am taking the PR out of Draft status.

**Comment by hernanmarino** [comment]
Updated to remove a (no lonnger valid) TODO comment as requested in https://github.com/bitcoin/bitcoin/pull/29428#issuecomment-1952480346. Also updated the PR title and description.

**Comment by hernanmarino** [comment]
> Please delete the hidden comment in the pull description, because this will end up in the merge commit.

Oh, sorry that. Deleted.

**Review by alfonsoromanz — APPROVED** [review]
Code review and tested ACK 4922f56660ae7834598c6dc4904a6711c23edfa3

File: test/functional/feature_assumeutxo.py [hunk 1]
```diff
@@ -20,8 +20,6 @@
       bad other serialization)
 - TODO: Valid snapshot file, but referencing a snapshot block that turns out to be
       invalid, or has an invalid parent
-- TODO: Valid snapshot file and snapshot block, but the block is not on the
-      most-work chain
```
[the remark this answers is not in the document]
Comment (alfonsoromanz) [reply test/functional/feature_assumeutxo.py:24]: Maybe the line that should be removed here is this one? `- TODO: A descendant of the snapshot block`. I believe the scenario where the snapshot block is not on the most-work chain is addressed here: https://github.com/bitcoin/bitcoin/pull/29996

[the remark this answers is not in the document]
Comment (hernanmarino) [reply test/functional/feature_assumeutxo.py:24]: > Why is this removed? The block is in the most-work chain. I think this comment meant a block that is _not_ in the most-work chain, no?

Maybe i missinterpreted the TODO comment meaning . To which TODO were you referring here  
https://github.com/bitcoin/bitcoin/pull/29428#issuecomment-1952480346 ?

[the remark this answers is not in the document]
Comment (hernanmarino) [reply test/functional/feature_assumeutxo.py]: Restored that TODO in my last push. Thanks

**Review by alfonsoromanz — APPROVED** [review]
Re ACK 2f1b1eee8b8a8f546fb5975ac529c4032e46f069

**Comment by hernanmarino** [comment]
Updated to re add an incorrectly removed TODO comment.

**Review by rkrux — APPROVED** [review]
crACK, utACK [2f1b1eee](https://github.com/bitcoin/bitcoin/pull/29428/commits/2f1b1eee8b8a8f546fb5975ac529c4032e46f069)

Thanks for adding the test, it's short and to the point. However, I've not been able to build the branch because it fails on my system (MAC i7) with the following error. I understand it is because this branch doesn't contain the latest changes of the master branch, notably this one: https://github.com/bitcoin/bitcoin/pull/29081.

```
util/time.cpp:54:9: error: use of undeclared identifier 'gmtime_s'; did you mean 'gmtime_r'?
    if (gmtime_s(&epoch, &time_val) != 0) {
        ^~~~~~~~
        gmtime_r
```

**Comment by achow101** [comment]
Needs rebase (for some reason Drahtbot isn't detecting this)

**Comment by hernanmarino** [comment]
> Needs rebase (for some reason Drahtbot isn't detecting this)

Rebased. Also deleted a couple of redundant comment as suggested by @maflcko to retrigger the false negative CI

File: test/functional/feature_assumeutxo.py [hunk 2]
```diff
@@ -161,6 +161,14 @@ def test_invalid_file_path(self):
         path = node.datadir_path / node.chain / "invalid" / "path"
         assert_raises_rpc_error(-8, "Couldn't open file {} for reading.".format(path), node.loadtxoutset, path)
 
+    def test_snapshot_with_less_work(self, dump_output_path):
+        self.log.info("Test bitcoind should fail when snapshot has less accumulated work than this node.")
+        node = self.nodes[0]
+        assert_equal(node.getblockcount(), FINAL_HEIGHT)
+        with node.assert_debug_log(expected_msgs=["[snapshot] activation failed - work does not exceed active chainstate"]):
+            assert_raises_rpc_error(-32603, "Unable to load UTXO snapshot", node.loadtxoutset, dump_output_path)
+        self.restart_node(0, extra_args=self.extra_args[0])
```
Comment (furszy) [inline test/functional/feature_assumeutxo.py:170]: why restart if you aren't doing anything with the node?

**Comment by kevkevinpal** [comment]
utACK [df6dc2a](https://github.com/bitcoin/bitcoin/pull/29428/commits/df6dc2aaaeffc664006b86ee8c8797dc484ec40e)

**Review by alfonsoromanz — APPROVED** [review]
Re ACK df6dc2aaaeffc664006b86ee8c8797dc484ec40e. Make is successful and the test passes. 

Screenshot 2024-05-28 at 09 55 15
Screenshot 2024-05-28 at 09 58 39

**Comment by achow101** [comment]
ACK df6dc2aaaeffc664006b86ee8c8797dc484ec40e

**Pull request merged**